src/webgpu/api/operation/adapter/requestAdapterInfo.spec.ts

```diff
@@ -0,0 +1,55 @@
+export const description = `
+Tests various ways of calling GPUAdapter.requestAdapterInfo.
+
+TODO:
+- Find a way to perform tests with and without user activation
+`;
+
+import { Fixture } from '../../../../common/framework/fixture.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { getGPU } from '../../../../common/util/navigator_gpu.js';
+import { assert } from '../../../../common/util/util.js';
+
+export const g = makeTestGroup(Fixture);
+
+const normalizedIdentifierRegex = /^$|^[a-z0-9]+(-[a-z0-9]+)*$/;
+
+g.test('adapter_info')
+  .desc(
+    `
+  Test that calling requestAdapterInfo with no arguments:
+    - Returns a GPUAdapterInfo structure
+    - Every member in the structure except description is properly formatted`
+  )
+  .fn(async t => {
+    const gpu = getGPU();
+    const adapter = await gpu.requestAdapter();
+    assert(adapter !== null);
+
+    const adapterInfo = await adapter.requestAdapterInfo();
+    assert(adapterInfo !== null);
```

Comment:
since the types don't allow this to be null, this isn't necessary. (if this did somehow return null, the test would catch it by crashing when you try to access the members)
```suggestion
```